### PR TITLE
Run test suite in parallel by default

### PR DESCRIPTION
**Pull request by enolan**
This is a lot faster on my machine. It outputs a bunch of "idris: bind: resource busy (Address already in use)" lines but they're irrelevant since AFAIK the tests don't actually connect to the socket.

### CODE REVIEW

**Comment by ahmadsalim** [comment]
@enolan It would still be nice to avoid the `idris: bind: resource busy` warnings in this PR, nonetheless. Would it be possible for you to look into that?

**Comment by enolan** [comment]
I'll make it so you can pass `--port none` and have it not bind a socket.

**Comment by ahmadsalim** [comment]
@enolan That would be good, thanks!

**Comment by ahmadsalim** [comment]
Although please include that in another PR!

**Comment by david-christiansen** [comment]
Is there anything stopping merging this as-is? Parallel tests would be very nice!

**Comment by enolan** [comment]
Rebased this on #3396 and set the flag in the interactive tests.

**Comment by enolan** [comment]
I reproduced the failure with the node backend on a different test:

```
    records001:        FAIL (10.26s)
      --- test/records001/output	2016-08-30 22:26:41.599501735 -0700
      +++ test/records001/expected	2016-07-11 11:47:11.472826733 -0700
      @@ -1,5 +1,5 @@
      -"Fred"
       "foo"
      +"Fred"
       [1, 2, 3]
       ["b", "a"]
       25
```
And another one:
```
    io002:             FAIL (7.40s)
      --- test/io002/output	2016-08-30 22:42:39.714194978 -0700
      +++ test/io002/expected	2016-07-11 11:47:11.380821293 -0700
      @@ -1,3 +1,3 @@
      -2
      -First and second? 1
      +First and second? 2
      +1
       2
```

Some sort of non-determinism is creeping in. Any ideas? I know basically nothing about Node.js or the backend.

**Comment by melted** [comment]
That happens sometimes with node. So it isn't new.

**Comment by ahmadsalim** [comment]
@enolan Would you please now, rebase this PR on the latest `master`? Thanks!

**Comment by enolan** [comment]
It's mergeable now

File: test/TestRun.hs [hunk 1]
```diff
@@ -104,11 +110,15 @@ runTest path flags = do
...
-    askOption $ \(NodeOpt node) ->
-      let (codegen, flags) = if node then (JS, ["--codegen", "node"])
-                                     else (C , [])
-       in
-        mkGoldenTests (testFamiliesForCodegen codegen)
-                    (flags ++ idrisFlags)
+main = do
+  -- The Node.js tests are nondeterministically wrong when run in parallel for
+  -- some reason, so we turn off parallelism when --node is passed. There's no
+  -- way to change paralellism after the test runner has started, so we parse
```
Comment (jfdm) [inline test/TestRun.hs]: Sorry to be pedantic, however, parallelism has been spelled incorrectly in its second use.

**Comment by enolan** [comment]
I removed the whole thing that turns off parallelism if we're using the Node backend, since the nondeterministic failures are fixed now. No more spelling error :)

**Comment by enolan** [comment]
Is there a reason this hasn't been merged?

**Comment by melted** [comment]
Guess everyone is waiting for someone else to do it. I'll break the deadlock.

**Pull request merged**